### PR TITLE
Single thread token fetching simulation

### PR DESCRIPTION
**Pull request by y-zeng**
Simulate the token fetching process without creating a new thread.

### CODE REVIEW

File: src/core/lib/security/credentials/fake/fake_credentials.c [hunk 1]
```diff
@@ -111,11 +119,15 @@ static void md_only_test_get_request_metadata(
   grpc_md_only_test_credentials *c = (grpc_md_only_test_credentials *)creds;
 
   if (c->is_async) {
-    grpc_credentials_metadata_request *cb_arg =
+    simulated_token_fetch_args *cb_arg =
+        gpr_malloc(sizeof(simulated_token_fetch_args));
```
Comment (jboeuf) [inline src/core/lib/security/credentials/fake/fake_credentials.c]: sizeof(*cb_arg)

File: src/core/lib/security/credentials/fake/fake_credentials.c [hunk 2]
```diff
@@ -111,11 +119,15 @@ static void md_only_test_get_request_metadata(
...
+    cb_arg->md_request =
         grpc_credentials_metadata_request_create(creds, cb, user_data);
-    grpc_executor_push(
-        grpc_closure_create(on_simulated_token_fetch_done, cb_arg),
-        GRPC_ERROR_NONE);
+    grpc_timer_init(exec_ctx, &cb_arg->timer,
+                    gpr_time_add(gpr_now(GPR_CLOCK_MONOTONIC),
+                                 gpr_time_from_micros(100, GPR_TIMESPAN)),
```
Comment (jboeuf) [inline src/core/lib/security/credentials/fake/fake_credentials.c]: It seems to me that this should be a constant.

File: src/core/lib/security/credentials/fake/fake_credentials.c (hunk 1 above)
Comment (y-zeng) [inline src/core/lib/security/credentials/fake/fake_credentials.c]: Fixed.

File: src/core/lib/security/credentials/fake/fake_credentials.c (hunk 2 above)
Comment (y-zeng) [inline src/core/lib/security/credentials/fake/fake_credentials.c]: Added macro to define this constant. Also set the time to 1 second.

**Comment by y-zeng** [comment]
Thanks a lot for the review! As discussed offline, it might be good to use the same approach as used in other tests.

**Pull request closed without merging**